zproject/backends.py

```diff
@@ -143,7 +144,14 @@ def get_or_create_user(self, username, ldap_user):
         try:
             return get_user_profile_by_email(username), False
         except UserProfile.DoesNotExist:
-            return UserProfile(), False
+            realm = Realm.objects.get(domain=settings.LDAP_APPEND_DOMAIN)
+            full_name = ldap_user.attrs[settings.AUTH_LDAP_USER_ATTR_MAP["full_name"]]
+            short_name = ldap_user.attrs[settings.AUTH_LDAP_USER_ATTR_MAP["full_name"]]
```

Comment:
Stylistically, I'd probably write this instead as an 

short_name = full_name

with a comment about how full_name is the intended fallback in this case, for less code duplication.

But actually, I think that for the short name here, it might be better to use the LDAP username as a fallback rather than the full name, since the whole point is to have something short to use for referring to the user...